engine/src/main/java/com/datatorrent/stram/plan/logical/LogicalPlan.java

```diff
@@ -1248,22 +1248,25 @@ public int hashCode()
   @Override
   public <T extends Operator> T addOperator(String name, T operator)
   {
-    if (operators.containsKey(name)) {
-      if (operators.get(name).operator == operator) {
-        return operator;
+    if ( name != null && !name.isEmpty()) {
```

Comment:
Check name and throw an exception. The rest of the method should not be affected.